File: src/17-challenge/solution.py
def detect_bombs(grid: list[list[bool]]) -> list[list[int]]:
    adjacents_bombs: list[list[int]] = []

    for i, row in enumerate(iterable=grid):
        bombs_in_row: list[int] = []

        for j, cell in enumerate(iterable=row):
            start_i: int = i - 1
            stop_i: int = i + 2

            start_j: int = j - 1
            stop_j: int = j + 2

            bombs = 0

            for _i in range(start_i, stop_i):
                if _i < 0 or _i > len(grid) - 1:
                    continue

                for _j in range(start_j, stop_j):
                    if _j < 0 or _j > len(row) - 1:
                        continue
                    bombs += grid[_i][_j]

            bombs_in_row.append(bombs - cell)

        adjacents_bombs.append(bombs_in_row)

    return adjacents_bombs

File: src/17-challenge/solution.js
/**
 * @param {boolean[][]} grid
 * @returns {number[][]}
 */
export default function detectBombs(grid) {
	/** @type {number[][]}>} */
	const adjacentsBombs = []

	for (let i = 0; i < grid.length; i++) {
		/** @type {boolean[]}>} */
		const row = grid[i]

		/** @type {number[]}>} */
		const bombsInRow = []

		for (let j = 0; j < row.length; j++) {
			/** @type {number}>} */
			const startI = i - 1

			/** @type {number}>} */
			const stopI = i + 2

			/** @type {number}>} */
			const startJ = j - 1

			/** @type {number}>} */
			const stopJ = j + 2

			/** @type {number}>} */
			let bombs = 0

			for (let _i = startI; _i < stopI; _i++) {
				if (_i < 0 || _i > grid.length - 1) continue

				for (let _j = startJ; _j < stopJ; _j++) {
					if (_j < 0 || _j > row.length - 1) continue
					bombs += ~~grid[_i][_j]
				}
			}

			bombsInRow.push(bombs - ~~row[j])
		}

		adjacentsBombs.push(bombsInRow)
	}

	return adjacentsBombs
}
